gtsfm/two_view_estimator.py

```diff
@@ -26,20 +27,67 @@
 pil_logger.setLevel(logging.INFO)
 
 
+# In case an epipolar geometry can be verified, it is checked whether
+# the geometry describes a planar scene or panoramic view (pure rotation)
+# described by a homography. This is a degenerate case, since epipolar
+# geometry is only defined for a moving camera. If the inlier ratio of
+# a homography comes close to the inlier ratio of the epipolar geometry,
+# a planar or panoramic configuration is assumed.
+# Based on COLMAP's front-end logic here:
+#    https://github.com/colmap/colmap/blob/dev/src/estimators/two_view_geometry.cc#L230
+MAX_H_INLIER_RATIO = 0.8
+
+EPSILON = 1e-6
+
+
+@dataclass(frozen=False)
+class TwoViewEstimationReport:
+    """
+    Args:
+        num_inliers_est_model: #correspondences consistent with estimated model (not necessarily "correct")
+        inlier_ratio_est_model: measures how polluted the putative matches were
+        num_inliers_gt_model: measures how well the verification worked, w.r.t. GT
+        inlier_ratio_gt_model: Only defined if GT relative pose provided
+        R_error_deg: relative pose error. Only defined if GT poses provided
+        U_error_deg
+    """
+
+    v_corr_idxs: np.ndarray
+    num_inliers_est_model: float
+    inlier_ratio_est_model: Optional[float] = None  # TODO: make not optional (pass from verifier)
+    num_inliers_gt_model: Optional[float] = None
+    inlier_ratio_gt_model: Optional[float] = None
+    R_error_deg: Optional[float] = None
+    U_error_deg: Optional[float] = None
+    i2Ri1: Optional[Rot3] = None
+    i2Ui1: Optional[Unit3] = None
+
+
 class TwoViewEstimator:
     """Wrapper for running two-view relative pose estimation on image pairs in the dataset."""
 
-    def __init__(self, matcher: MatcherBase, verifier: VerifierBase, corr_metric_dist_threshold: float) -> None:
+    def __init__(
+        self,
+        matcher: MatcherBase,
+        verifier: VerifierBase,
+        eval_threshold_px: float,
+        estimation_threshold_px: float,
+        min_num_inliers_acceptance: int,
+    ) -> None:
         """Initializes the two-view estimator from matcher and verifier.
 
         Args:
             matcher: matcher to use.
             verifier: verifier to use.
-            corr_metric_dist_threshold: distance threshold for marking a correspondence pair as inlier.
+            eval_threshold_px: distance threshold for marking a correspondence pair as inlier during evaluation (not estimation).
+            estimation_threshold_px: distance threshold for marking a correspondence pair as inlier during estimation
```

Comment:
Should that not be called inlier_threshold_px? Why do we need different thresholds for eval vs estimation?

Reply:
So for evaluation, we might want to be very strict, e.g. look for cases where we get matches under 2 px in epipolar distance.

But for fitting, we might want a different threshold, e.g. 4 px, like in COLMAP.

This is actually what we have already been doing before.

Reply:
Yep this is what we should do